### ISSUE TITLE
`@stylistic/indent`: unexpected error for semicolon in 5.2.0 release

### ISSUE THREAD

**Issue by Eugeno** [eslint-stylistic/eslint-stylistic#901]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```
type SomeType =
  'one'
  | 'two'
  | 'four'
; // Expected indentation of 2 spaces but found 0
```

### Reproduction

https://stackblitz.com/edit/github-rdkk4twr-ac7p67dj?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
